edx_name_affirmation/api.py

```diff
@@ -0,0 +1,110 @@
+"""
+Python API for edx_name_affirmation.
+"""
+
+import logging
+
+from edx_name_affirmation.models import VerifiedName
+
+log = logging.getLogger(__name__)
+
+
+def create_verified_name(
+    user, verified_name, profile_name, verification_attempt_id=None,
+    proctored_exam_attempt_id=None, is_verified=False,
+):
+    """
+    Create a new `VerifiedName` for the given user.
+
+    Arguments:
+        * `user` (User object)
+        * `verified_name` (str): Representative of the name on the user's physical ID card.
+        * `profile_name` (str): A snapshot of either 1) the user's pending name change if given
+          or 2) the existing name on the user's profile.
+        * `verification_attempt_id` (int): Optional reference to an external ID verification
+          attempt.
+        * `proctored_exam_attempt_id` (int): Optional reference to an external proctored exam
+          attempt.
+        * `is_verified` (bool): Optional, defaults False. This should determine whether the
+          verified_name is valid for use with ID verification, exams, etc.
+    """
+    # Do not allow empty strings
+    if verified_name == '':
+        err_msg = _get_empty_string_err_msg(
```

Comment:
NIT: I might use a custom exception for these to make this more readable. It seems fine either way, but just something to consider